torch/_dynamo/variables/misc.py

```diff
@@ -767,16 +767,15 @@ def func(tx, item, acc):
             return variables.ListIteratorVariable(
                 items, mutable_local=MutableLocal(), **options
             )
-        elif (
-            self.value is functools.wraps
-            and not kwargs
-            and len(args) == 1
-            and args[0].source
-        ):
+        elif self.value is functools.wraps and not kwargs and len(args) == 1:
 
             def wraps(fn):
                 if isinstance(fn, variables.NestedUserFunctionVariable):
-                    return fn.clone(wraps_source=args[0].source)
+                    # `wraps_source` should implement `reconstruct`
+                    # which NestedUserFunctionVariable does.
+                    # We prefer `args[0].source` if that is available.
+                    wraps_source = args[0].source if args[0].source else fn
+                    return fn.clone(wraps_source=wraps_source)
```

Comment:
@kshitij12345 IMO we should always just pass one of them to be consistent. Do things break if we always pass NestedUserFunctionVariable?

Reply:
Have made an update, passing `fn` was the wrong thing to do.